services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -146,6 +146,11 @@ async def remove_users_manually_marked_as_guests(
                 app=app, project_uuid=project_uuid, user_id=guest_user_id,
             )
 
+        # if there are not projects, just remove the user
+        await remove_resources_if_guest_user(
```

Comment:
If I understand it well, if ``remove_resources_if_guest_user`` is called with a valid guest user and a project, it deletes the project and then then user ( see ``delete_user`` line 222 )?!!

Assume a GUEST user with multiple projects. The first iteration will delete the first project and the user. The second iteration will detect that the user is not GUEST (i.e. does not exists) and will return leaving the succesive projects undeleted

Reply:
Almost correct. This function has evolved badly over time.

It was first designed to remove GUESTS which had one project. Now it is used in 3 separate places. Where it has to behave in 3 different ways. This is bad design and should be changed.

Usages of the function:
- user has exactly 1 project (its main purpose)
- user has more then one project (that's why the user removal was try cached)
- user has no projects (that's what these changes try to address)

Reply:
To properly implement this I would need to add the following tests
```
# TODO: tests for garbage collector
# - a User with more then 2 projects
# - a user without projects
# - a user with just 1 project
#
#  The user can be:
# - connected via browser (websocket connection is up)
# - disconnected (no websocket connection)
```
I would need a hand with how to simulate the users and projects in the database.